relative-path/src/lib.rs

```diff
@@ -290,26 +290,48 @@
 
 #![allow(clippy::manual_let_else)]
 #![deny(missing_docs)]
+#![no_std]
 
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
+#[cfg(feature = "std")]
+extern crate std;
+
+#[cfg(feature = "std")]
+use std::prelude::v1::*;
+
+#[cfg(feature = "std")]
 mod path_ext;
 
 #[cfg(test)]
 mod tests;
 
+#[cfg(feature = "std")]
 pub use path_ext::{PathExt, RelativeToError};
 
-use std::borrow::{Borrow, Cow};
-use std::cmp;
-use std::error;
-use std::fmt;
-use std::hash::{Hash, Hasher};
-use std::iter::FromIterator;
-use std::mem;
-use std::ops;
-use std::path;
-use std::rc::Rc;
-use std::str;
-use std::sync::Arc;
+#[cfg(feature = "alloc")]
+use alloc::borrow::{Borrow, Cow, ToOwned};
+#[cfg(feature = "alloc")]
+use alloc::boxed::Box;
+#[cfg(feature = "alloc")]
+use alloc::rc::Rc;
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+#[cfg(feature = "alloc")]
+use alloc::sync::Arc;
+use core::cmp;
+use core::fmt;
+use core::hash::{Hash, Hasher};
+#[cfg(feature = "alloc")]
+use core::iter::FromIterator;
+use core::mem;
+#[cfg(feature = "alloc")]
+use core::ops;
+use core::str;
```

Comment:
nit: The order I group imports of `core`, `alloc`, `std`.

Reply:
I was using the order that rustfmt uses, which is alphabetical.
But happy to use a different order :-)